app/javascript/hyacinth_ui_v1/components/digital_objects/new/AssetNew.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Form } from 'react-bootstrap';
+import { useMutation } from '@apollo/react-hooks';
+import uploadFile from '../../../graphql/file_uploads';
+import { createAssetMutation } from '../../../graphql/assets';
+
+function debugObject(obj) {
+  console.log(obj);
+  return new Promise(resolve => resolve(obj));
+}
+
+function AssetNew(props) {
+  const { parentId, refetch } = props;
+  const [createAsset] = useMutation(createAssetMutation);
+  const elementId = `input-${String.toString(Math.random()).replace('.', '')}`;
+
+  const onChange = (event) => {
+    const {
+      target: {
+        validity,
+        files: [file],
+      },
+    } = event;
+    if (!validity.valid) return false;
+    function resetFileInput() { document.getElementById(elementId).value = ''; }
+    const url = `/api/v1/digital_objects/${parentId}/uploads`;
+    return uploadFile(file, url)
```

Comment:
I got bound up here on the REST errors vs GraphQL errors thing - I think it would be much easier to implement after the Rails 6 change, but if that looks like it's a long way off I'll work something out.

Reply:
That makes sense, we can definitely address this later.